File: backend/models/Room.js
const mongoose = require('mongoose');

const roomSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true
  },
  propertyId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Property',
    required: true
  },
  type: {
    type: String,
    enum: ['LIVING_ROOM', 'BEDROOM', 'KITCHEN', 'BATHROOM', 'OFFICE', 'GARAGE', 'OTHER'],
    required: true
  },
  deviceIds: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Device'
  }]
}, {
  timestamps: true
});

// Index for efficient queries
roomSchema.index({ propertyId: 1 });

module.exports = mongoose.model('Room', roomSchema); 

File: ios/Models/Room.swift
import Foundation

struct Room: Codable, Identifiable {
    let id: String
    let name: String
    let propertyId: String
    let type: RoomType
    var deviceIds: [String]?
    
    enum RoomType: String, Codable {
        case livingRoom = "LIVING_ROOM"
        case bedroom = "BEDROOM"
        case kitchen = "KITCHEN"
        case bathroom = "BATHROOM"
        case office = "OFFICE"
        case garage = "GARAGE"
        case other = "OTHER"
    }
} 